src/ImageSharp/Processing/Processors/PixelShading/Abstract/PixelShaderProcessorBase.cs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Six Labors and contributors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.Numerics;
+
+using SixLabors.ImageSharp.Advanced;
+using SixLabors.ImageSharp.Advanced.ParallelUtils;
+using SixLabors.ImageSharp.PixelFormats;
+using SixLabors.Primitives;
+
+namespace SixLabors.ImageSharp.Processing.Processors.PixelShading.Abstract
+{
+    /// <summary>
+    /// Applies a user defined pixel shader effect through a given delegate.
+    /// </summary>
+    /// <typeparam name="TPixel">The pixel format.</typeparam>
+    internal abstract class PixelShaderProcessorBase<TPixel> : ImageProcessor<TPixel>
+        where TPixel : struct, IPixel<TPixel>
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PixelShaderProcessorBase{TPixel}"/> class.
+        /// </summary>
+        /// <param name="source">The source <see cref="Image{TPixel}"/> for the current processor instance.</param>
+        /// <param name="sourceRectangle">The source area to process for the current processor instance.</param>
+        protected PixelShaderProcessorBase(Image<TPixel> source, Rectangle sourceRectangle)
+            : base(source, sourceRectangle)
+        {
+        }
+
+        /// <inheritdoc/>
+        protected override void OnFrameApply(ImageFrame<TPixel> source)
+        {
+            var interest = Rectangle.Intersect(this.SourceRectangle, source.Bounds());
+            int startX = interest.X;
+
+            ParallelHelper.IterateRowsWithTempBuffer<Vector4>(
+                interest,
+                this.Configuration,
+                (rows, vectorBuffer) =>
+                {
+                    for (int y = rows.Min; y < rows.Max; y++)
+                    {
+                        Span<Vector4> vectorSpan = vectorBuffer.Span;
+                        int length = vectorSpan.Length;
+                        Span<TPixel> rowSpan = source.GetPixelRowSpan(y).Slice(startX, length);
+                        PixelOperations<TPixel>.Instance.ToVector4(this.Configuration, rowSpan, vectorSpan);
```

Comment:
Done in https://github.com/SixLabors/ImageSharp/pull/1065/commits/ed02d3525a7f667871dcf0c20c8e5e165289eebb. Followed the same convention we used for the bokeh blur parameters, since you mentioned you don't want default parameters in public APIs 👍